src/picasso-def/components/treemap.js

```diff
@@ -158,7 +158,7 @@ export const treemap = () => ({
             };
             rects.push(childRect);
           }
-        } else if (node.depth === 2 && treeHeight > 1 && node.data.label) {
+        } else if (node.depth === 2 && treeHeight > 1 && node.data.label && labels.headers && labels.overlay) {
```

Comment:
I think this shoud be `(labels.headers && labels.overlay || labels.auto)`

Reply:
I feel that labels.auto is not needed here as whenever auto is true, overlay can never be, and in this condition we are only populating overlay nodes.

Reply:
If you have auto and 3 (or more) dimension, it should show header and overlays

Reply:
That case works as it is handled in the above if statement. Screenshot below : Three Dimensions - Auto mode
![Screenshot 2023-01-30 at 15 30 07](https://user-images.githubusercontent.com/30860406/215505049-b6263a5a-a3dd-4d57-bdc9-26f0cb6430a7.png)

Reply:
but [multi-scenario-5-baseline.png](https://github.com/qlik-oss/sn-treemap/pull/111/files#diff-fba4c752fba79175830c91c7311cb5412f90631e99e4ac00e209c1623be72a89) is 3 dim with auto labels
and should keeps the overlay labels but don't

Reply:
This is strange because I have specifically tested this scenario and as seen in the screenshot above, it works.

Reply:
Shall have a look !